workbench-core/example/app/src/routes/datasetRoutes.ts

```diff
@@ -73,12 +73,13 @@ export function setUpDSRoutes(
         throw Boom.badRequest('datasetid request parameter is invalid');
       }
       processValidatorResult(validate(req.body, CreateExternalEndpointSchema));
-      await dataSetService.addDataSetExternalEndpoint(
-        req.params.datasetId,
-        req.body.externalEndpointName,
-        dataSetStoragePlugin,
-        req.body.externalRoleName
-      );
+      await dataSetService.addDataSetExternalEndpointForUser({
+        dataSetId: req.params.datasetId,
+        externalEndpointName: req.body.externalEndpointName,
+        storageProvider: dataSetStoragePlugin,
+        userId: req.body.userId, // TODO get authenticated user instead?
```

Comment:
Your TODO seems pretty trivial today, less trivial tomorrow. Probably best to either switch to `res.locals.user` or just delete the comment.

Reply:
I wasnt sure if the idea behind the route was to add an endpoint for the calling user or for a user as defined by the request body. Any preferences?

Reply:
We will be getting the `userId` from the authenticated user